### DISCUSSION TITLE
Fundamentos Orientacao a objetos (Mao na massa - notificacoes e heranca)

### DISCUSSION THREAD

**Discussion by LuisTerranova** [balta-io/forum#682] [Tira Dúvidas]
Entao, nessa aula eu nao entendi o motivo de utilizar a classe Base. O balta justificou como uma base para todas as outras, mas a classe Content ja nao fazia isso? por que foi necessaria uma segunda classe "base"? No caso da aula ele so tirou o id de Content e colocou na Base... Ficou confuso pra mim. Agora todos tem IDs e herdam de base de alguma forma, mas nao entendi o motivo

**Comment by andrebaltieri** [comment]
Bom dia, @LuisTerranova, como vai?

A ideia foi justamente reusar o `Id` em todas as entidades, não apenas na Content... mas eu poderia sim derivar de Content ao invés de criar uma classe base.

A herança traz todo conteúdo de uma classe A para uma classe B! Este conceito vai ficando mais claro a medida que for avançando nas carreiras.

Veja se ficou alguma dúvida e me retorna aqui por favor!
